tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Process/ProcessStartCommon.cs

```diff
@@ -67,9 +71,164 @@ internal static Scope CreateScope(string filename, IDictionary<string, string> e
+                    {
+                        finalCommandExec.Add(string.Empty);
+                        continue;
+                    }
+
+                    var truncatedArg = Truncate(arg, MaxCommandLineLength, out truncated);
+                    finalCommandExec.Add(truncatedArg);
+                    size -= truncatedArg.Length;
+
+                    if (size <= 0)
+                    {
+                        truncated = true;
+                    }
+                }
+
+                // Serialized as JSON array string because tracer only supports string values
+                tags.CommandExec = JsonConvert.SerializeObject(finalCommandExec);
+                tags.Truncated = truncated ? "true" : null;
+            }
+
+            return tags;
+        }
+
+        private static string Truncate(string value, int maxLength, out bool truncated)
+        {
+            truncated = false;
+            if (string.IsNullOrEmpty(value) || value.Length <= maxLength)
+            {
+                return value;
+            }
+
+            truncated = true;
+            return value.Substring(0, maxLength);
+        }
+
+        private static List<string> SplitStringIntoArguments(string input)
+        {
+            var result = new List<string>();
+            var currentArgument = string.Empty;
+            var inSingleQuotes = false;
+            var inDoubleQuotes = false;
+            var escapeNextCharacter = false;
+
+            foreach (var currentChar in input)
+            {
+                if (escapeNextCharacter)
+                {
+                    currentArgument += currentChar;
+                    escapeNextCharacter = false;
+                }
+                else if (currentChar == '\\')
+                {
+                    escapeNextCharacter = true;
+                }
+                else if (currentChar == '"' && !inSingleQuotes)
+                {
+                    inDoubleQuotes = !inDoubleQuotes;
+                    if (!inDoubleQuotes)
+                    {
+                        result.Add(currentArgument);
+                        currentArgument = string.Empty;
+                    }
+                }
+                else if (currentChar == '\'' && !inDoubleQuotes)
+                {
+                    inSingleQuotes = !inSingleQuotes;
+                    if (!inSingleQuotes)
+                    {
+                        result.Add(currentArgument);
+                        currentArgument = string.Empty;
+                    }
+                }
+                else if (currentChar == ' ' && !inSingleQuotes && !inDoubleQuotes)
+                {
+                    if (currentArgument.Length > 0)
+                    {
+                        result.Add(currentArgument);
+                        currentArgument = string.Empty;
+                    }
+                }
+                else
+                {
+                    currentArgument += currentChar;
```

Comment:
For this kind of operations, using string builders instead of strings can give you a better performance.

Reply:
Yes, good idea thanks 😄